primitives/io/src/batch_verifier.rs

```diff
@@ -139,14 +137,12 @@ impl BatchVerifier {
 		self.sr25519_items.clear();
 
 		if pending.len() > 0 {
-			let pair = Arc::new((Mutex::new(false), Condvar::new()));
-			let pair_clone = pair.clone();
-
+			let (sender, receiver) = std::sync::mpsc::channel();
```

Comment:
I still don't understand why all this code isn't just:
```
futures::executor::block_on(futures::future::join_all(pending))
```

Reply:
Integration tests (`test_sync`) are panicking, they use it and you can't have one executor inside another.